### PR TITLE
Add test for earth_radiius_mod.F90, fixed some code warnings in other tests

### LINKED ISSUE #117: test for earth_radius_mod.F90
There are no tests for this mod.

### PR DESCRIPTION
**Pull request by edwardhartnett**
Add test for earth_radiius_mod.F90, fixed some code warnings in other tests.

Fixes #117 
Part of #108

### CODE REVIEW

File: tests/test_earth_radius.F90 [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  if (abs(radius - 6378160) > epsilon(radius)) stop 6
+  if (abs(eccen_squared - 6.7226700223333219E-003) > epsilon(eccen_squared)) stop 7
+
+  ! Case 3.
+  igdtmpl(1) = 2
+  call earth_radius(igdtmpl, igdtlen, radius, eccen_squared)
+  if (abs(radius - 6378160) > epsilon(radius)) stop 8
+  if (abs(eccen_squared - 6.7226700223333219E-003) > epsilon(eccen_squared)) stop 9
+
+  ! Case 3.
```
Comment (GeorgeGayno-NOAA) [inline tests/test_earth_radius.F90]: Check your 'Case' numbering.

File: tests/test_earth_radius.F90 [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  ! Case 0.
+  igdtmpl(1) = 0
+  call earth_radius(igdtmpl, igdtlen, radius, eccen_squared)
+  if (abs(radius - 6367470.0) > epsilon(radius)) stop 2
+  if (abs(eccen_squared - 0.0) > epsilon(eccen_squared)) stop 3
+
+  ! Case 1.
+  igdtmpl(1) = 1
+  igdtmpl(2) = 1
+  igdtmpl(3) = 1
```
Comment (GeorgeGayno-NOAA) [inline tests/test_earth_radius.F90:22]: igdtmpl(3) contains octets 17-20, which holds the scale value of the radius of a spherical earth.  I think your example produces an earth radius of 1 meter. I know these are idealized tests. But I might pick a more realistic earth radius.

File: tests/test_earth_radius.F90 (hunk 1 above)
Comment (edwardhartnett) [reply tests/test_earth_radius.F90]: Thanks!

File: tests/test_earth_radius.F90 (hunk 2 above)
Comment (edwardhartnett) [reply tests/test_earth_radius.F90:22]: Good idea!

**Comment by edwardhartnett** [comment]
@GeorgeGayno-NOAA please approve if you have no additional comments.

Actually the reason for this test file is I want to try out a solution for the 4, 8, d problem, and the earth_radius_mod is simple enough to use for a prototype to see if it works...

**Review by GeorgeGayno-NOAA — APPROVED** [review]

**Pull request merged**